### ISSUE TITLE
Training for ACE Dataset

### ISSUE THREAD

**Issue by nishantgurunath** [dwadden/dygiepp#13]
Hi,
When are you going to release the code for training on ACE dataset? 
Thanks

**Comment by dwadden** [comment]
Sorry for dragging my feet, I've been working on other projects. I'll try to get it released this weekend.

**Comment by tkuculo** [comment]
Any progress? Also, will you release a pre-trained model for the ACE dataset?

**Comment by dwadden** [comment]
EMNLP arXiv preprint deadline is next Sat. I should be able to do it after that. Sorry... other projects.

**Comment by edchengg** [comment]
Do you mind share the exact 22 argument role labels you use in your paper for ace2005 event extraction? thank you!

**Comment by dwadden** [comment]
I've updated the README to train ACE relation and event models. The event training works properly and a pre-trained model is available for download. there's still something weird going on with the relation training and we're debugging, but at least there's a script for you to run if you want to.

**Comment by dwadden** [comment]
@edchengg 

```
Place
Entity
Attacker
Victim
Destination
Person
Artifact
Target
Instrument
Agent
Defendant
Origin
Recipient
Giver
Org
Adjudicator
Buyer
Vehicle
Plaintiff
Seller
Prosecutor
Beneficiary
```

**Comment by plscoggers** [comment]
I noticed there's a pretrained model for ACE05 event but not relation.  Is there a pretrained model for relation or is being held back due to the issue?

**Comment by dwadden** [comment]
Yep I wasn't going to release until we sort out the low performance, but I can go ahead and release them if that would be useful.

**Comment by plscoggers** [comment]
Yeah it would be interesting to at least look at.  Though I haven't seen any mention of what the issue is.  Is it something to do with going from tensorflow to pytorch?

**Comment by dwadden** [comment]
It's not even a TF / PyTorch thing. I just lost track of the training config that gave the best performance, and the model is fairly sensitive to learning rate, batch size, etc. I'll try to get something up in the next few days though.

**Comment by dwadden** [comment]
OK, I finally released an ACE relation model. It gets roughly 54 F1, which is worse than the 62 in the paper but still usable. It's got to be an issue with the training config. Pull requests that get better performance are welcome!

Closing this.
